src/luxonis_ml/embeddings/emb_duplicate.py

```diff
@@ -0,0 +1,273 @@
+        The image_paths of the most similar embeddings.
+
+    """
+    # Get the reference embeddings
+    # check if reference_embeddings is a list of instance_ids
+    if isinstance(reference_embeddings, str):
+        reference_embeddings = [reference_embeddings]
+    if isinstance(reference_embeddings[0], str):
+        reference_embeddings = get_embeddings_from_ids(qdrant_client, reference_embeddings, collection_name=collection_name)
+
+    # Select the reference embedding
+    if method == 'first':
+        if isinstance(reference_embeddings, list):
+            reference_embeddings = np.array(reference_embeddings)
+        if len(reference_embeddings.shape) > 1:
+            reference_embeddings = reference_embeddings[0]
+    elif method == 'average':
+        # Calculate the average of the reference embeddings
+        avg_embedding = np.mean(reference_embeddings, axis=0)
+        reference_embeddings = avg_embedding
+    else:
+        raise ValueError(f'Unknown method: {method}')
+    
+    ix, vals, res = search_qdrant(qdrant_client, reference_embeddings, collection_name=collection_name, data_name=dataset, limit=n)
+    ix, similarities, res = np.array(ix), np.array(vals), np.array(res)
+    
+    # Select the best k embeddings
+    if k_method is None:
+        best_embeddings_ix = np.argsort(similarities)[-k:]
+        
+    elif k_method == 'kde_basic':
+        _, new_min, _, _ = kde_peaks(similarities, bandwidth=kde_bw, plot=plot)
+        
+        if len(new_min) > 0:
+            k = len(np.where(similarities > new_min[-1])[0])
+            if k < 2 and len(new_min) > 1:
+                k = len(np.where(similarities > new_min[-2])[0])
+            print(k, new_min)
+
+        best_embeddings_ix = np.argsort(similarities)[-k:]
+
+    elif k_method == 'kde_peaks':
+
+        if len(similarities) > 50000:
+            print('Too many embeddings, using 97 percentile')
+            # take top 97 procentile of closest points
+            p_97 = np.percentile(similarities, 97)
+            ix_sim = np.where(similarities > p_97)[0]
+            _, new_min, _, _ = kde_peaks(similarities[ix_sim], bandwidth=kde_bw, plot=plot)
+            
+            if len(new_min) > 0:
+                k = len(np.where(similarities > new_min[-1])[0])
+                print(k, new_min)
+
+        else:
+            # get maxima and minima of the KDE on the distances
+            _, minima, _, _ = kde_peaks(similarities, bandwidth=kde_bw, plot=plot)
+            if len(minima) > 0:
+                minima = minima[-1]
+                if minima < 0.94:
+                    minima = 0.97
+                else:
+                    minima = max(minima, 0.96)
+                
+            else:
+                minima = 0.97
+
+            # select k closest embeddings
+            k = len(np.where(similarities > minima)[0])
+            if minima > 0.97 and k < 2:
+                k = len(np.where(similarities > 0.97)[0])
+            print(k, minima)
```

Comment:
There are a lot of constants here. I think it's fine to keep them as is for now if it seems to work well, but we might want to experiment with changing them and if they should be changed by a user in the future.

Reply:
I think it would also make sense to move model_utils and qdrant_utils to new /utils subdir, then add constant.py and define like RELEVANT_THRESHOLD_NAME?